testcontainers/src/core/container.rs

```diff
@@ -168,20 +168,30 @@ where
             })
     }
 
-    /// Returns the bridge ip address of docker container as specified in NetworkSettings.IPAddress
+    /// Returns the bridge ip address of docker container as specified in NetworkSettings.Networks.IPAddress
     pub fn get_bridge_ip_address(&self) -> IpAddr {
         IpAddr::from_str(
             &self
                 .docker_client
                 .inspect(&self.id)
                 .network_settings
-                .unwrap_or_default()
-                .ip_address
+                .and_then(|settings| {
```

Comment:
Instead of continuing to nest here, please panic early. Makes the control flow much easier to understand!

Reply:
panic? here was `unwrap_or_defaul` so I saved this behavior and `and_then` won't apply anything for `Err` value (so even if we would use panic, it also has fail-fast semantic). It's idiomatic rust code. The only reason I can see - panic with different messages on different levels

It isn't easier to understand to me, but I definitely can change, so I will use rules in this repo

Reply:
@thomaseizinger I refactored it a bit with more clear flow & error messages, pls take a look